crates/core/tedge_actors/src/builders.rs

```diff
@@ -305,3 +310,141 @@ impl<Req: Message, Res: Message> Builder<ConcurrentServerMessageBox<Req, Res>>
         self.build_concurrent()
     }
 }
+
+pub struct Concurrent;
+pub struct Sequential;
```

Comment:
Enum variants are runtime things. Here, one needs to distinguish at compile time the kind of actor to build. One could also use explicit type annotations when calling the `build()` method - but that's not so easy as this method is used by the runtime which has no insight on which variant to build. Another option could be to use phantom types, however using these *tag* types seems more natural (at least to my eyes).

Reply:
Yeah, I kinda got the feeling that you wanted something static at compile time. And yes, I'd also prefer these explicit types rather than `Phantom` types to keep things simple.